ui/src/main/java/edu/wpi/grip/ui/codeGeneration/java/TSocket.java

```diff
@@ -0,0 +1,56 @@
+package edu.wpi.grip.ui.codegeneration.java;
+
+/**
+ * Created by Toby on 5/26/16.
+ */
+public class TSocket {
+  protected String type;
+  protected String name;
+  private static final String MutableOf = "MutableOf<";
+  public TSocket(String type, String name){
+    this(type);
+    this.name = name;
+  }
+
+  public TSocket(String type){
+    setType(type);
+  }
+  public void setType(String type){
+	  if(type.equals("Integer")||type.equals("Double")||type.equals("Boolean")||type.equals("Number")){
+	    	this.type = MutableOf+type+">";
+	    }
+	  else{
+		  this.type = type;
+	  }
+  }
+  public String type(){
+    return type;
+  }
+
+  public String name(){
+    return name;
+  }
+
+  public String baseType(){
+	 if(!mutable()){	
+		 return type;
+	 }
+	 else{
+		 String retVal = type.replace(MutableOf, "").replace(">", "");
```

Comment:
That would be a better way to do it than it was done in that version. However the java classes no longer will be responsible for keeping track of MutableOf since that is java specific and these classes are meant to exist for all three languages.